### PR TITLE
36 admin invoice show page update invoice status

### LINKED ISSUE #30: 36. Admin Invoice Show Page: Update Invoice Status
As an admin     
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
I see the invoice status is a select field
And I see that the invoice's current status is selected
When I click this select field,
Then I can select a new status for the Invoice,
And next to the select field I see a button to "Update Invoice Status"
When I click this button
I am taken back to the admin invoice show page
And I see that my Invoice's status has now been updated

### PR DESCRIPTION
**Pull request by pcbennett108**
#### Updates Made -
- User Stories Updated: US 36
- Database/Migrations: N/A
- Routes: add :update to admin/invoices
- Models: N/A
- Controllers: add update method to invoices
- Views: add invoice status update form to admin/invoices show page
- Testing:
  - Models: un-skipped 2nd test for disabled merchants on line 238 in merchant_spec
  - Features: add block to test invoice status update in invoices/show
- Considerations(Extra notes, refactors, etc.): Closes #30

### CODE REVIEW

File: spec/models/merchant_spec.rb [hunk 1]
```diff
@@ -235,7 +235,7 @@
       expect(Merchant.enabled).to eq([@merchant_1, @merchant_2, @merchant_3, @merchant_4])
     end
 
-    xit "returns all disabled merchants" do
+    it "returns all disabled merchants" do
```
Comment (Antoine-Aube) [inline spec/models/merchant_spec.rb:238]: hopefully this time around we get this right! Not sure what's going on with this persistance. Maybe we just push from main if this doesn't do the trick.

Comment (pcbennett108) [reply spec/models/merchant_spec.rb:238]: I was thinking the same thing.

File: spec/features/admin/invoices/show_spec.rb [hunk 2]
```diff
@@ -69,5 +69,21 @@
...
     end
+
+    it "US 36 has a for to update the invoice status" do
+      customer_1 = Customer.create!(first_name: "Bob", last_name: "Smith")
+      invoice_1 = Invoice.create!(status: 1, customer_id: customer_1.id, created_at: Time.new(2011,4,5))
+
+      visit admin_invoice_path(invoice_1)
+
+      expect(page).to have_select(:status, with_options: ["in progress", "completed", "cancelled"], selected: "completed")
+      expect(page).to have_button("Update Invoice Status")
```
Comment (Antoine-Aube) [inline spec/features/admin/invoices/show_spec.rb:80]: The only thing I would add here is an expectation that the invoice has the status, like expect(invoice_1.status.to eq("completed"), then once you update, expect the update to change that status with another expect statement. Might be redundant, bit tests for the database aspect of things.

File: config/routes.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@
   
   namespace :admin do 
     resources :merchants, except: [:destroy]
-    resources :invoices, only: [:index, :show]
+    resources :invoices, only: [:index, :show, :update]
```
Comment (Antoine-Aube) [inline config/routes.rb:17]: this is great, makes adding routes so much easier!

File: app/controllers/admin/invoices_controller.rb [hunk 4]
```diff
@@ -6,4 +6,10 @@ def index
   def show
     @invoice = Invoice.find(params[:id])
   end
+
+  def update
+    invoice = Invoice.find(params[:id])
+    invoice.update(status: params[:status])
```
Comment (Antoine-Aube) [inline app/controllers/admin/invoices_controller.rb:12]: super solid and clean!

File: app/views/admin/invoices/show.html.erb [hunk 5]
```diff
@@ -1,6 +1,13 @@
 <%= render partial: "admin/page_head" %>
 
 <center><h3>Invoice #<%= @invoice.id %></h3></center>
+
+<%= form_with url: admin_invoice_path(@invoice), method: :patch, data:{turbo: false} do |f| %>
```
Comment (Antoine-Aube) [inline app/views/admin/invoices/show.html.erb:5]: Looks great. For a refactor, we could change this to a model form form the Async lesson.

File: spec/features/admin/invoices/show_spec.rb (hunk 2 above)
Comment (pcbennett108) [reply spec/features/admin/invoices/show_spec.rb:80]: Yes, that's a good point.

**Pull request merged**